contrib/init/bitcoind.service

```diff
@@ -12,12 +12,11 @@ Description=Bitcoin daemon
 After=network.target
 
 [Service]
-ExecStart=/usr/bin/bitcoind -daemon -conf=/etc/bitcoin/bitcoin.conf -pid=/run/bitcoind/bitcoind.pid
+ExecStart=/usr/bin/bitcoind -daemon=0 -conf=/etc/bitcoin/bitcoin.conf -pid=/run/bitcoind/bitcoind.pid
 # Creates /run/bitcoind owned by bitcoin
 RuntimeDirectory=bitcoind
 User=bitcoin
-Type=forking
-PIDFile=/run/bitcoind/bitcoind.pid
+Type=simple
```

Comment:
In commit "init: Don't daemonize when runnning under systemd" (558dddfa8a493a3fa521d7bd098ec6d87e148ced)

I don't think changing from a forking service to a simple service is actually a good idea, and I wonder if there any downsides at all to just dropping this commit, while keeping the other two commits, because:

- Simple services are more limited that forking services. You can't tell the difference between a running service and service that's starting up, and you can't do things like specifying Requires= and After= in a dependent service to avoid starting it if starting bitcoind fails.

- Using a simple service requires passing `-daemon=0` which currently has the side effect of enabling logging to stdout. If any debug options are specified this can mean a lot of information will go into system logs that isn't going there now. Also, since bitcoind prints timestamps on the console by default, the log entries will look strange in journalctl.

It might be a good idea to add `-printtoconsole=0` if we need to keep `-daemon=0`, or at least mention in release notes that new logs may now show up in journald.

Reply:
You are right! After doing some digging I think `forking` is superior right now. Perhaps the following changes can be made in future PRs:

1. Use `sd_notify` to give `systemd` more information on the state of the process if a compile flag `--enable-systemd-notify` is supplied to `./configure`
2. Make logs available in `journald` in a sane way somehow